src/server/service/diagnostics.rs

```diff
@@ -1000,7 +1005,7 @@ mod log {
 
         // for rotated *.<rotated-datetime> file
         if let Some(res) = filename.strip_prefix((log_file.to_owned() + ".").as_str()) {
-            if NaiveDateTime::parse_from_str(res, DATETIME_ROTATE_SUFFIX).is_ok() {
+            if NUM_REGEX.is_match(res) {
```

Comment:
Seems `regex` is unnecessary.

Reply:
```rust
if let Some(nums) = res.get_mut(0..4) {
	if nums.iter().all(|&x| x.is_numeric()); {
		return true;
	}
}
```
Is ok?

Reply:
What's wrong with original check?

Reply:
```
tikv.log.2020-04-07-00:29:00 // old
tikv.log2020-04-07-00:29:00.488218178 // new
```

The older version rotating file naming can not be match by `DATETIME_ROTATE_SUFFIX`, so we modify the check logic to `^\d{4}`.

Reply:
But now a dot is added, the original check should pass now?

Reply:
It checks milliseconds previously, which is not compatible with 3.0.x rotated files, so that the checker is also relaxed.